cider/tests/functionality/CiderStringTest.cpp

```diff
@@ -348,6 +348,35 @@ TEST_F(CiderStringNullableTestArrow, ArrowSubstringTest) {
   assertQueryArrow("SELECT SUBSTRING(col_2, -4, 2) FROM test ");
 }
 
+TEST_F(CiderStringNullableTestArrow, ArrowCaseConvertionTest) {
+  // SELECT LOWER(column) FROM table
+  assertQueryArrow("SELECT col_2, LOWER(col_2) FROM test;", "stringop_lower_null.json");
+
+  /// NOTE: (YBRua) Skipped for now because we dont expect queries without FROM clauses.
+  /// 1. Behaviors of Cider and DuckDb are different w.r.t. this query.
+  ///    DuckDb produces only 1 row, while Cider produces input_row_num rows.
+  ///    Because the compiled row_func IR always runs for input_row_num times
+  ///    at runtime in current implementation of Cider.
+  /// 2. If no input table (no FROM clause) is given, the generated Substrait plan will
+  ///    have a "virtualTable" (instead of a "namedTable") as a placeholder input.
+  ///    <https://substrait.io/relations/logical_relations/#virtual-table>
+  // SELECT LOWER(literal)
+  // assertQueryArrow("SELECT LOWER('ABCDEFG');");
+
+  // SELECT LOWER(literal) FROM table
+  assertQueryArrow("SELECT LOWER('aAbBcCdD12') FROM test;",
+                   "stringop_lower_literal_null.json");
+
+  // SELECT UPPER(column) FROM table
+  assertQueryArrow("SELECT col_2, UPPER(col_2) FROM test;", "stringop_upper_null.json");
```

Comment:
I think we can add some case like `SELECT * FROM test WHERE UPPER(col_2) = "AAA" ` once #144 got merged.

Reply:
added